src/lib/shared/signature.ts

```diff
@@ -0,0 +1,42 @@
+import { babyjubjub } from "babyjubjub-ecdsa";
+
+export const generateSignatureKeyPair = (): {
+  signingKey: string;
+  verifyingKey: string;
+} => {
+  const keyPair = babyjubjub.ec.genKeyPair();
+
+  const pubKey = keyPair.getPublic();
+  const privKey = keyPair.getPrivate();
+
+  return {
+    verifyingKey: pubKey.encode("hex"),
+    signingKey: privKey.toString("hex"),
+  };
+};
+
+export const sign = (signingKey: string, data: string): string => {
+  const key = babyjubjub.ec.keyFromPrivate(signingKey, "hex");
+  const msgHash = BigInt(
+    "0x" + babyjubjub.ec.hash().update(data).digest("hex")
+  );
+  const signature = key.sign(msgHash.toString(16), "hex", {
+    canonical: true,
+  });
+  const signatureDER = signature.toDER();
+
+  return Buffer.from(signatureDER).toString("base64");
+};
+
+export const verify = (
```

Comment:
Is there a reason this doesn't call [verifyEcdsaSignature](https://github.com/jubmoji/babyjubjub-ecdsa/blob/c4c7c0a346644d6f9d802d5fef538f0e92fc7eea/packages/lib/src/ecdsa.ts#L15)? This is the verification used to verify ARX signatures, so it would be good to ensure compatibility.